src/components/base/BaseSelect.vue

```diff
@@ -14,14 +15,25 @@ const props = withDefaults(
     errorMessage: "",
   }
 );
-defineEmits(["update:modelValue", "blur"]);
+const emit = defineEmits<{
+  (e: "update:modelValue", id: string | number): void;
+  (e: "blur"): void;
+}>();
 
 const selectClasses = computed(() => {
   return [
     "base-select__select",
     props.errorMessage ? "base-select__select--error" : "",
   ];
 });
+const handleEmit = (event: Event) => {
+  emit(
+    "update:modelValue",
+    (event.target as HTMLInputElement).value.length > 1
+      ? (event.target as HTMLInputElement).value
+      : parseInt((event.target as HTMLInputElement).value)
```

Comment:
`(event.target as HTMLInputElement).value` is repetitive. could be first assigned to e.g. `const target`